prepare_release.sh

```diff
@@ -32,18 +35,20 @@ fi
 
 GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
 if [[ ! $GIT_BRANCH =~ $UPSTREAM_BRANCH ]]; then
-    echo "You do not seem to be on the ${UPSTREAM_BRANCH} branch!"
+    echo "You do not seem to be on the $UPSTREAM_BRANCH branch!"
     exit 1
 fi
 
 if [ -n "$(git status --porcelain)" ]; then
     echo "You have local changes!"
-    exit 1
+    if [[ $DRY_RUN = 0 ]]; then
```

Comment:
Why do we allow local changes for dry runs?
Those changes would be thrown away if one follows the instructions to clean up after the dry run.

Reply:
I found it quite convenient to be able to modify the script and do dry runs without the need to remove this check or commit changes all the time.

The script commits only the changes in package.json to the release branch and the clean up instructions should only throw away the release branch